File: app/src/main/java/com/dgd/earthquakes/ui/splash/SplashActivity.java
package com.dgd.earthquakes.ui.splash;

import android.support.v4.app.Fragment;

import com.dgd.earthquakes.injection.Injection;
import com.mdgd.commons.support.v7.fragment.HostActivity;

/**
 * Created by Max
 * on 01-May-17.
 */

public class SplashActivity extends HostActivity<SplashScreenContract.IPresenter> implements SplashScreenContract.IView {

    @Override
    protected SplashScreenContract.IPresenter getPresenter() {
        return Injection.getSplashPresenter(this);
    }

    @Override
    protected Fragment getFirstFragment() {
        return SplashFragment.newInstance();
    }
}

File: app/src/main/java/com/dgd/earthquakes/ui/splash/SplashActivity.kt
package com.dgd.earthquakes.ui.splash

import android.os.Bundle

import com.dgd.earthquakes.Injection
import com.dgd.earthquakes.common.FragmentHostActivity
import com.dgd.earthquakes.common.HostedFragment
import com.dgd.earthquakes.ui.quakes.MainHostActivity

/**
 * Created by Max
 * on 01-May-17.
 */

class SplashActivity : FragmentHostActivity<SplashScreenContract.IPresenter>(), SplashScreenContract.IView {

    override fun getFirstFragment(savedInstanceState: Bundle?): HostedFragment<*> {
        presenter.updateQuakes()
        return SplashFragment.newInstance()
    }

    override fun setupPresenter(): SplashScreenContract.IPresenter {
        return Injection.getSplashPresenter(this)
    }

    override fun proceedFromSplash() {
        startActivity(MainHostActivity.getIntent(this@SplashActivity))
    }

    override fun showProgress(title: String, msgResId: Int) {
        showProgress(title, getString(msgResId))
    }
}
